vars/updatecli.groovy

```diff
@@ -40,19 +35,55 @@ def call(userConfig = [:]) {
         org.jenkinsci.plugins.pipeline.modeldefinition.Utils.markStageSkippedForConditional(updatecliRunStage)
       }
     }
+
     stage(updatecliRunStage) {
       if (runUpdatecli) {
-        withCredentials([
-          usernamePassword(
-          credentialsId: finalConfig.credentialsId,
-          usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
-          passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
-          )
-        ]) {
-          sh 'updatecli version'
-          sh updatecliCommand
-        } // withCredentials
-      } // if (runUpdateCli)
-    } // stage
-  }
+        withEnv(["PATH+CUSTOM=$customUpdatecliPath"]) {
+          // Wrap entire execution inside PATH update
+          // If a custom version is provided, download and install that version of updatecli
+          if (finalConfig.version) {
+            stage("Download updatecli version ${finalConfig.version}") {
+              sh 'echo $PATH'
+              withEnv(["UPDATECLI_VERSION=${finalConfig.version}", "CUSTOM_UPDATECLI_PATH=${customUpdatecliPath}"]) {
+                // Only custom installation vars
+                sh '''
+                                    cpu="$(uname -m)"
+                                    # Normalize CPU architecture for Updatecli release filenames
+                                    if [ "$cpu" = "aarch64" ] || [ "$cpu" = "arm64" ]; then
+                                        cpu="arm64"
+                                    fi
+                                    # Construct the tar file name based on the CPU architecture.
+                                    tarFileName="updatecli_Linux_${cpu}.tar.gz"
+                                    downloadUrl="https://github.com/updatecli/updatecli/releases/download/v${UPDATECLI_VERSION}/${tarFileName}"
+                                    curl --silent --show-error --location --output ${tarFileName} ${downloadUrl}
```

Comment:
```suggestion
                                    curl --silent --show-error --location --output ${tarFileName} "https://github.com/updatecli/updatecli/releases/download/v${UPDATECLI_VERSION}/${tarFileName}"
```

No need for intermediate variable. The debug mode of the shell will prints the variable if you need it :)